xbmc/interfaces/json-rpc/SettingsOperations.cpp

```diff
@@ -684,6 +695,26 @@ bool CSettingsOperations::SerializeSettingAddon(const CSettingAddon* setting, CV
   return true;
 }
 
+bool CSettingsOperations::SerializeSettingDate(const CSettingDate* setting, CVariant &obj)
```

Comment:
It shouldn't be too hard to write the setting definitions in JSON instead of XML and write a parser for JSON. Now (after all the GUI integration rewriting) it would probably even be possible to write such a parser outside of the whole settings library. That's basically what I did for the backwards compatibility layer. Maybe it would be worth extracting all the deserialization logic from the settings library and leave that up to the "integrator". There might be some tricky business around being able to overwrite setting definitions e.g. per platform but it should all be doable.